### DISCUSSION TITLE
integrate in mpp-solar EASUN INVERTERS

### DISCUSSION THREAD

**Discussion by Tolmino** [jblance/mpp-solar#494] [General]
hi, i have bought a ISOLAR-SMX-II-5.6KP-WIFI from EASUN INVERTERS that i would read with mpp-solar.
i have tried all protocols and i have obtained nothing.
the first think i have noticed is that i can read ttyusb and not hidraw0 device, 
because doing lsusb -d 1a36:7523 i can read that this inverter has a HL-340 USB-Serial adapter from QinHeng Electronics....
so i have asked to EASUN to give me the protocol datasheet...
i attach it here....
[常规英文--储能逆变器MODBUS监控协议寄存器地址表V1.5.pdf](https://github.com/jblance/mpp-solar/files/15412137/--.MODBUS.V1.5.pdf)

**Comment by Saentist** [comment]
@Tolmino documentation say "MODBUS communication protocol"
your "usb-to-rs232" is not compatible. need "usb-to-rs485"
